src/NetAnalyzers/VisualBasic/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/BasicOverrideEqualsOnOverloadingOperatorEquals.vb

```diff
@@ -63,8 +63,9 @@ Namespace Microsoft.CodeQuality.VisualBasic.Analyzers.ApiDesignGuidelines
         Private Shared Function HasEqualityOperator(type As INamedTypeSymbol) As Boolean
             For Each member In type.GetMembers()
                 Dim method = TryCast(member, IMethodSymbol)
-                If method?.MethodKind = MethodKind.UserDefinedOperator AndAlso
-                    CaseInsensitiveComparison.Equals(method.Name, WellKnownMemberNames.EqualityOperatorName) Then
+                If method IsNot Nothing AndAlso
+                   method.MethodKind = MethodKind.UserDefinedOperator AndAlso
+                   CaseInsensitiveComparison.Equals(method.Name, WellKnownMemberNames.EqualityOperatorName) Then
```

Comment:
I'm not following. If `method` is null, shouldn't `method?.MethodKind = MethodKind.UserDefinedOperator` evaluate to false and the second `AndAlso` operand is not evaluated?

To me `method?.MethodKind = MethodKind.UserDefinedOperator` is equivalent to `method IsNot Nothing AndAlso method.MethodKind = MethodKind.UserDefinedOperator`, but I must be missing something.

Reply:
Well, I thought the same, but apparently not. Below executable throws an NRE when executed with no command line arguments:

```vb
Imports System

Module Program
    Sub Main(args As String())
        Console.WriteLine("Hello World!")

        Dim c As C = Nothing
        If (args.Length <> 0) Then
            c = New C
            c.FieldA = A.F1
        End If

        If c?.FieldA = A.F1 AndAlso Equals(c.ToString(), "ConsoleApp54.C") Then
            Console.WriteLine("True")
        Else
            Console.WriteLine("False")
        End If
    End Sub
End Module

Class C
    Public FieldA As A
End Class

Public Enum A
    F1
    F2
End Enum
```

_VB never ceases to surprise!_

Reply:
That's very nice to know!